### PR TITLE
Disallow cloned kustomization from using a local base outside the clone dir.

### PR DESCRIPTION
**Pull request by monopole**
Prevent attack where victim has somehow stored a hostile kustomization directory under /tmp, and then download a kustomization that specifies, as a base, a relative path to that hostile kustomization.

Unlikely, as kustomizations are downloaded to /tmp/kustomize-{randomString}, and deleted after the build finishes, but there might be some other angle on this, and the use case lost here makes little sense.

### CODE REVIEW

File: pkg/loader/fileloader.go [hunk 1]
```diff
@@ -166,6 +166,10 @@ func (l *fileLoader) New(path string) (ifc.Loader, error) {
 		}
 		return newGitLoader(path, l.fSys, l.referrer, l.cloner)
 	}
+	if l.uri != "" {
```
Comment (Liujingfang1) [inline pkg/loader/fileloader.go]: I think this restriction is too strong. I will also block the remote bases from accessing transitive bases in the same repo.

Comment (liggitt) [reply pkg/loader/fileloader.go]: transitive based in the same repo definitely seems useful

Comment (monopole) [reply pkg/loader/fileloader.go]: yes, the way we track the "repo's dir" is a bit of a mess at the moment.
I can continue in this PR, or check this in, and loosen the restriction in a subsequent PR.

Comment (monopole) [reply pkg/loader/fileloader.go]: meh, as long as nobody is panicking, i'll loosen it here.  will ping later.

**Comment by monopole** [comment]
Don't review yet, just wanted to upload a checkpoint. :)

**Pull request merged**